### PR TITLE
add example of passing a struct as an arg to a signal

### LINKED ISSUE #283: Add examples for passing struct arg via Signal
As per https://community.temporal.io/t/deserialization-error-for-signal-struct/1609/2; there is a need to have a more complete example for passing in a struct via signal.

Should I add an example here? --> https://github.com/temporalio/documentation/blob/master/docs/go-signals.md

### PR DESCRIPTION
**Pull request by vkarpov15**
Fix #283

## What does this PR do?

Add an example of passing a struct to a signal and decoding it using [mapstructure](https://github.com/mitchellh/mapstructure)

### CODE REVIEW

File: docs/go/signals.md [hunk 1]
```diff
@@ -72,6 +72,39 @@ if err != nil {
...
 
+## Signalling with structs
+
+You can also send structs in signals, as long as the struct is [serializable](https://pkg.go.dev/go.temporal.io/sdk/converter#CompositeDataConverter.ToPayload).
+`Receive()` decodes data into a generic map, to decode the value into a struct you should use a library like [mapstructure](https://github.com/mitchellh/mapstructure).
+Note that Temporal only serializes public fields.
+
+```go
+// Make sure all fields you want to serialize are public. Temporal
+// serializes `Message`, not `message`.
```
Comment (tsurdilo) [inline docs/go/signals.md:83]: > Temporal serializes 'Message', not 'message'

Could this comment be generalized to mention how this applies for any struct field? (serialized "as-is"/"preserving the case"/....)

**Review by sw-yx — APPROVED** [review]
useful! thank you

File: docs/go/signals.md [hunk 2]
```diff
@@ -72,6 +72,39 @@ if err != nil {
 
 You can find more example Signals usage in our [Go Samples](https://github.com/temporalio/samples-go).
 
+## Signalling with structs
+
+You can also send structs in signals, as long as the struct is [serializable](https://pkg.go.dev/go.temporal.io/sdk/converter#CompositeDataConverter.ToPayload).
```
Comment (flossypurse) [inline docs/go/signals.md:77]:
```suggestion
You can also send structs in Signals, as long as the struct is [serializable](https://pkg.go.dev/go.temporal.io/sdk/converter#CompositeDataConverter.ToPayload).
```

File: docs/go/signals.md [hunk 3]
```diff
@@ -72,6 +72,39 @@ if err != nil {
 
 You can find more example Signals usage in our [Go Samples](https://github.com/temporalio/samples-go).
 
+## Signalling with structs
```
Comment (flossypurse) [inline docs/go/signals.md:75]:
```suggestion
## Signal with structs
```

Just trying to avoid gerunds in titles when possible.

**Review by flossypurse — APPROVED** [review]

**Pull request merged**